Use shared threshold includes in trigger and condition doc templates (#3209)
home-assistant/developers.home-assistant: docs/documenting/integration-docs-examples.md

The **Light brightness changed** trigger fires after...
Use it to...

{% include integrations/labs_entity_triggers_note.md %}

{% include triggers/ui_header.md %}

{% include triggers/threshold_changed_steps.md
   title="Light brightness changed"
   sensor="light"
   areas="living room or bedroom"

>    unit_phrase_ui="a fixed percentage (0–100%)" %}

Reviewer: @abmantis I'm hesitant to change this here, because it's meant as a generic trigger example, which happens to work with that modular include.

In general, I couldn't make up my mind how I feel about this include in general - while the current copy-pasted block is big, it's also possible to review in the PR diff. Reviewing these will require always looking at the deploy app to check if the input is used correctly in the sentences.

tl;dr: I'd rather reuse content that's self-contained, so it doesn't have to be reviewed each time.

@c0ffeeca7 @Laufmaschine WDYT?

Author: @fdmarcin When reviewing, isn't it easier to miss mistakes in the big copy-pasted text blocks, than in this smaller include? I do understand the argument about not being to see the rendered result directly, though. 

I'll wait for your team's decision before changing anything else here or close it.

Reviewer: @abmantis Thanks! I guess both are true 😅 

I haven't seen opposition to this idea in general (@Laufmaschine?). Would you mind editing the PR to have this treshhold-based example as a suboption, so this example stays generic?

Reviewer 2: Hi! I really like the single-source content approach but I understand the concerns from @fdmarcin. If we change the template, I suppose that then we will need to apply the approach to all the triggers and conditions. Questions that cross my mind:
- How much effort will this mean?
- Is this a priority right now, considering the added value for the end user? It will of course grant content consistency across all pages and it will be easier to update in the future.
- Will this make the documentation editing by contributors harder in any way and discourage them to do so?

tl;dr: I vote in a generic example with no include here, at the moment.

Author: > - How much effort will this mean?

I think you and the docs team can answer this much better than me, since you are the ones that know how much effort it was to review all the new triggers and conditions. Assuming those will continue to grow, the effort will increase. Although LLM can do the changes easily nowadays, review might still be annoying with a lot of files changed.

> - Is this a priority right now, considering the added value for the end user? It will of course grant content consistency across all pages and it will be easier to update in the future.

Except for the consistency, I see the value here more for the developers/writers/maintainers, since it makes doc changes smaller.

> - Will this make the documentation editing by contributors harder in any way and discourage them to do so?

I would say it makes it easier, because you have a small include snippet to use instead of having to copy/paste and review a bunch of text sections.

Regarding the consistency, when I added the templates I noticed that we already diverging triggers or conditions. I don't recall which one it was, but at least one of the ones I've replaced with the include was outdated. 

But since @fdmarcin had mixed feelings about using the templates and @Laufmaschine prefers not to use them, then I'll close this one.

Reviewer: @abmantis Thank you for your work until now Abílio!

```

> ### Template: condition (threshold-based)

Reviewer: @abmantis I think this is the way to go here - have subsections for specific cases that we can cover with the include.

Reviewer: Should we change the existing "Template: trigger" to use a non-treshold one and then have another subsection (h4) like this one for conditions?

```suggestion
#### Template: condition (threshold-based)
```

> Some conditions test a numeric reading against a threshold (above, below, in range, or outside range) instead of a simple on/off state. These use the shared `conditions/threshold_value_*` includes. Adapt this template for that kind of condition:

Reviewer: To match https://github.com/home-assistant/developers.home-assistant/pull/3209/changes#r3537104633

```suggestion
Some conditions test a numeric reading against a threshold (above, below, in range, or outside range) instead of an on/off state.
For these conditions, use the shared `conditions/threshold_value_*` includes.

Adapt this template for the options sections in that kind of condition:
```

```md
---
title: "Light brightness"
condition: light.is_brightness
domain: light
description: "Tests if the brightness of one or more lights is above, below, or within a range."
related_conditions:
  - light.is_on
  - light.is_off
---

The **Light brightness** condition passes when...
Use it to...

> 
> {% include integrations/labs_entity_triggers_note.md %}

Reviewer: This is no longer needed:

```suggestion
```

{% include conditions/ui_header.md %}

{% include conditions/threshold_value_steps.md
   title="Light brightness"
   sensor="light"
   areas="living room or bedroom"
   value_long="a fixed percentage directly, for example `65` for 65%" %}

### Options in the UI

{% include conditions/threshold_value_options_ui.md
   value_short="a fixed percentage (0–100)" %}

{% include conditions/yaml_header.md %}

In YAML, refer to this condition as `light.is_brightness`. A basic example looks like this:

{% example %}
condition: |
  condition: light.is_brightness
  target:
    entity_id: light.living_room
  options:
    threshold:
      type: above
      value:
        number: 60
    behavior: any
{% endexample %}

This passes when the living room light's brightness is above 60%.

### Options in YAML

YAML sometimes provides additional options for more complex use cases that are not available through the UI.

{% include conditions/threshold_value_options_yaml.md
   range_note="0–100"
   number_final="a percentage value (0–100)" %}

<!-- Keep the two "includes" below if your integration supports targets -->
{% include conditions/targets.md %}

> {% include conditions/behavior.md %}
> 
> ## Good to know
> 
> - Add extra information here.
> 
> {% include conditions/try_it.md %}
> 
> {% include conditions/more_examples.md %}
> 
> ### Automation: pause a movie if the room is too bright
> 
> When you press play, only dim-check the room: if the living room light is brighter than 60%, turn it down first so the screen isn't washed out.
> 
> - **Trigger**: State: Media player starts playing
> - **Condition**: Light brightness (above 60%)
>   - **Target**: Living room light
>   - **Condition passes if**: Any
> - **Action**: Light: Turn on (brightness 30%)
> 
> {% details "YAML example for dimming a too-bright room on playback" %}
> 
> {% example %}
> automation: |
>   alias: "Dim living room when starting a movie"
>   triggers:
>     - trigger: state
>       entity_id: media_player.living_room
>       to: "playing"
>   conditions:
>     - condition: light.is_brightness
>       target:
>         entity_id: light.living_room
>       options:
>         threshold:
>           type: above
>           value:
>             number: 60
>         behavior: any
>   actions:
>     - action: light.turn_on
>       target:
>         entity_id: light.living_room
>       data:
>         brightness_pct: 30
> {% endexample %}
> 
> {% enddetails %}
> 
> {% include conditions/stuck.md %}
> 
> {% include conditions/related.md %}
> 

Reviewer: From here on, this is the same as the non-treshold example, so we can cut it and note it in the section intro:

```suggestion
{% include conditions/behavior.md %}

<!-- Continue with remaining sections, following "Template: condition" -->
```
